### PR TITLE
DAOS-11524 build: rpmlint resulting RPMs

### PR DESCRIPTION
**Pull request by brianjmurrell**

### CODE REVIEW

File: src/tests/ftest/cart/util/cart_logtest.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
```
Comment (phender) [inline src/tests/ftest/cart/util/cart_logtest.py:1]: Don't we need this for python files that we execute directly, like this one?

File: src/tests/ftest/scripts/main.sh [hunk 2]
```diff
@@ -1,4 +1,3 @@
-#!/bin/bash
```
Comment (phender) [inline src/tests/ftest/scripts/main.sh:1]: Isn't this still needed?

File: src/tests/ftest/scripts/setup_nodes.sh [hunk 3]
```diff
@@ -1,4 +1,3 @@
-#!/bin/bash
```
Comment (phender) [inline src/tests/ftest/scripts/setup_nodes.sh:1]: Isn't this still needed?

File: utils/completion/daos.bash [hunk 4]
```diff
@@ -1,4 +1,3 @@
-#!/bin/bash
```
Comment (phender) [inline utils/completion/daos.bash:1]: Isn't this still required?

File: Jenkinsfile [hunk 5]
```diff
@@ -13,7 +13,7 @@
 
 // To use a test branch (i.e. PR) until it lands to master
 // I.e. for testing library changes
-//@Library(value="pipeline-lib@your_branch") _
+@Library(value="pipeline-lib@bmurrell/rpm-linting") _
```
Comment (phender) [inline Jenkinsfile:16]: Holding out for final approval until this is no longer needed.

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/cart/util/cart_logtest.py (hunk 1 above)
Comment (brianjmurrell) [reply src/tests/ftest/cart/util/cart_logtest.py:1]: Do we actually _execute_ this one?  I can't see how we would given it's permissions are 644 -- so no execute bit set on it and I don't see the RPM packaging adding the execute permission on packaging it up.

File: src/tests/ftest/scripts/main.sh (hunk 2 above)
Comment (brianjmurrell) [reply src/tests/ftest/scripts/main.sh:1]: No.  This file is not executed.  It is included in a remote command, effetively with a `cat` (although it is a `sed` not a `cat` but the net result is the same): https://github.com/daos-stack/daos/blob/ecca8a53c23cec4f59da930701c91dede2b804b0/ftest.sh#L157

File: src/tests/ftest/scripts/setup_nodes.sh (hunk 3 above)
Comment (brianjmurrell) [reply src/tests/ftest/scripts/setup_nodes.sh:1]: Same story here: https://github.com/daos-stack/daos/blob/ecca8a53c23cec4f59da930701c91dede2b804b0/ftest.sh#L132

File: utils/completion/daos.bash (hunk 4 above)
Comment (brianjmurrell) [reply utils/completion/daos.bash:1]: Given the permissions on the file, I don't think so.  I think the bash completion stuff sources these files, and doesn't execute them directly.

**Comment by brianjmurrell** [comment]
Sorry, this should have been opened as a draft.  Changed.

File: src/tests/ftest/cart/util/cart_logtest.py (hunk 1 above)
Comment (brianjmurrell) [reply src/tests/ftest/cart/util/cart_logtest.py]: It seems we do execute it and the perms in my local repo got changed.  Not sure how.  In any case, I have reverted this one.

**Comment by brianjmurrell** [comment]
@phender, @JohnMalmberg, @raydennis This is still waiting for hardware testing but everything this PR changes has been tested in the current pipeline.  I just forgot to skip HW testing in the commit pragma, so this can be reviewed any time now.

**Review by JohnMalmberg — APPROVED** [review]

**Review by raydennis — DISMISSED** [review]

**Review by JohnMalmberg — DISMISSED** [review]

**Comment by brianjmurrell** [comment]
@daos-stack/daos-gatekeeper HW testing was skipped as it would have provided no additional testing benefit to this PR.

**Comment by brianjmurrell** [comment]
@JohnMalmberg @raydennis I lost a merge race so this needed a rebase and I figured I may as well put a small fix for a previous commit into this one since it has to go through CI again anyway.  If you could please re-review I'd appreciate it.   Thanks!

**Review by raydennis — DISMISSED** [review]

**Review by JohnMalmberg — DISMISSED** [review]

File: src/tests/ftest/util/verify_perms.py [hunk 6]
```diff
@@ -1,6 +1,5 @@
-#!/usr/bin/env python3
```
Comment (phender) [inline src/tests/ftest/util/verify_perms.py:1]: This file is executed as a command in tests, so this shouldn't be removed.

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/util/verify_perms.py (hunk 6 above)
Comment (brianjmurrell) [reply src/tests/ftest/util/verify_perms.py]: Done

**Review by raydennis — APPROVED** [review]

**Review by JohnMalmberg — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Pull request merged**